app.py

```diff
@@ -41,8 +42,8 @@ def get_data(name):
     )
 
 
-def generate_select_boxes(options_x, options_y, options_color, key_prefix):
-    select_boxes = [None, None, None]
+def generate_select_boxes(options_x, options_y, options_color_1, options_color_2, key_prefix):
+    select_boxes = [None, None, None, None]
```

Comment:
This function aims at reducing code duplication when creating the widgets of the different plots. If you want to introduce a widget specific to a plot (like `option_color_2`, only used for `plot_donut_complex`), a better practice would be to the selectbox outside of this function, just before calling `plot_donut_complex`. This would reduce the need to adapt the code everywhere we call `generate_select_boxes`.